### PR TITLE
retry file watcher errors

### PR DESCRIPTION
**Pull request by ramaraochavali**
This PR retries file watcher errors so that it helps to recover from transient errors.

[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[ X] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[X ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

Pull Request Attributes

Please check any characteristics that apply to this pull request. 

[ X] Does not have any changes that may affect Istio users.

### CODE REVIEW

**Comment by ramaraochavali** [comment]
@howardjohn I thought about this a bit more and I felt it is better to add retry so that we can recover from transient failures. In future we can tie this to liveness probe (adding a new one for agent) with a watcher error threshold so that if the errors are non transient we can restart the pod.

File: security/pkg/nodeagent/cache/secretcache.go [hunk 1]
```diff
@@ -327,16 +344,17 @@ func (sc *SecretManagerClient) addFileWatcher(file string, resourceName string)
...
 		cacheLog.Debugf("already watching file for %s", file)
 		// Already watching, no need to do anything
-		return
+		return nil
 	}
 	sc.fileCerts[key] = struct{}{}
 	// File is not being watched, start watching now and trigger key push.
 	cacheLog.Infof("adding watcher for file certificate %s", file)
 	if err := sc.certWatcher.Add(file); err != nil {
 		cacheLog.Errorf("%v: error adding watcher for file, skipping watches [%s] %v", resourceName, file, err)
```
Comment (howardjohn) [inline security/pkg/nodeagent/cache/secretcache.go:353]: nit: "skipping" log is wrong now.

**Review by howardjohn — APPROVED** [review]

**Pull request merged**